test/parallel/test-stream-writable-decoded-encoding.js

```diff
@@ -56,3 +56,27 @@ class MyWritable extends stream.Writable {
   m.write('some-text', 'utf8');
   m.end();
 }
+
+{
+  assert.throws(() => {
+    const m = new MyWritable(null, {
+      defaultEncoding: 'my invalid encoding',
```

Comment:
You should also add a test for passing null, and clarify if it should throw error or not.

Reply:
I'll add a test with null